Sources/TuistGenerator/Mappers/AutogeneratedProjectSchemeWorkspaceMapper.swift

```diff
@@ -5,12 +5,13 @@ import TuistGraph
 /// Mapper that generates a new scheme `ProjectName-Project` that includes all targets
 /// From a given workspace
 public final class AutogeneratedProjectSchemeWorkspaceMapper: WorkspaceMapping { // swiftlint:disable:this type_name
-    private let enableCodeCoverage: Bool
+    private var enableCodeCoverage: Bool { codeCoverageMode != nil }
+    private let codeCoverageMode: CodeCoverageMode?
 
     // MARK: - Init
 
-    public init(enableCodeCoverage: Bool) {
-        self.enableCodeCoverage = enableCodeCoverage
+    public init(codeCoverageMode: CodeCoverageMode?) {
+        self.codeCoverageMode = codeCoverageMode
     }
 
     public func map(workspace: WorkspaceWithProjects) throws -> (WorkspaceWithProjects, [SideEffectDescriptor]) {
```

Comment:
I'd extend the tests of this class to ensure the right code coverage targets are set.